InvenTree/web/templatetags/spa_helper.py

```diff
@@ -34,3 +34,9 @@ def spa_bundle():
     return mark_safe(
         f"""<script type="module" src="{settings.STATIC_URL}web/{index['file']}"></script>{imports_files}"""
     )
+
+
+@register.simple_tag
+def spa_settings():
+    """Render settings for spa."""
+    return mark_safe(f"""<script>window.INVENTREE_SETTINGS={json.dumps(settings.PUI_SETTINGS)}</script>""")
```

Comment:
Question: Can we move the serialisation step into global as the settings will not change during runtime?

Reply:
Into what global? As I discovered, to my suprise, these template tags only get called once after each server start for the first request.

Reply:
File globals so it gets loaded when Django start up. The tags probably get cached if you use the same client and inputs. The page also only gets loaded by the browser if you do a hard refresh or new tab - I tried to reduce that as much as possible to get away from django template rendering.

Reply:
where are these file globals?

Reply:
Everything in module level is global in Python ([longer description](https://realpython.com/python-scope-legb-rule/#modules-the-global-scope)). So if you move the json.dumps out of the function onto the same level as spa_settings it will only need be run once

Reply:
Oh yeah, now got what you want. I thought you wanted a global template tag or something what I didn't have knowledge on. Refactored now.